scripts/app.js

```diff
@@ -115,16 +166,79 @@ input.addEventListener("keyup", function() {
 });
 
 input.addEventListener("keydown", function (event) {
+    if (event.shiftKey && event.code === "Tab") {
+        event.preventDefault();
+
+        manageTabulation(-2);
+        return;
+    }
+
     // Tabulation
-    if (event.keyCode === 9) {
+    if (event.code === "Tab") {
+        event.preventDefault();
+
+        manageTabulation(2);
+    }
+
+    // Parenthesis
+    if (event.key === '(' || event.key === '[' || event.key === '{' || event.key === '"' || event.key === "'") {
+        const start = this.selectionStart;
+        const end = this.selectionEnd;
+
+        if (start === end) {
+            return;
+        }
         event.preventDefault();
-        var cursor = input.selectionStart;
-        insert(input, "  ");
-        input.selectionStart = cursor + 2;
-        input.selectionEnd = input.selectionStart;
+
+        const selectedText = this.value.substring(start, end);
+
+        const open = event.key;
+        const close = (event.key === '(') ? ')' : (event.key === '[') ? ']' : (event.key === '{') ? '}' : event.key;
+
+        // Set textarea value to: text before selection + open + selected text + close + text after selection
```

Comment:
Comment is useless and redondant with the code.

If you want, you can add named variables for `this.value.substring(0, start)` and `this.value.substring(end)`, and you'll obtain the same readibility as with your comment.